sunkit_image/conftest.py

```diff
@@ -84,3 +88,35 @@ def pytest_runtest_setup(item):
     if isinstance(item, pytest.Function):
         if "remote_data" in item.keywords and not HAVE_REMOTEDATA:
             pytest.skip("skipping remotedata tests as pytest-remotedata is not installed")
+
+
+@pytest.mark.remote_data
+@pytest.fixture(scope="session")
+def test_granule_map():
+    return sunpy.map.Map("https://github.com/sunpy/data/raw/main/sunkit-image/granule_testdata.fits")
+
+
+@pytest.fixture(scope="session")
+def test_granule_minimaps():
+
+    # array with "intergranule region"
+    arr_1 = np.ones((10, 10))
+    arr_1[0, 0] = 0 
+    # modified array with "intergranule region"
+    arr_2 = np.ones((10, 10))
+    arr_2[1, 1] = 0 
+    # array with no "intergranule region"
+    arr_3 = np.ones((10, 10))
+
+    observer = get_earth()
+    frame = Helioprojective(observer=observer, obstime=observer.obstime)
+    ref_coord = astropy.coordinates.SkyCoord(0, 0, unit='arcsec', frame=frame)
+    
+    header1 = make_fitswcs_header(arr_1, ref_coord,)
+    map1 =  sunpy.map.GenericMap(arr_1, header1)
+    header2 = make_fitswcs_header(arr_2, ref_coord,)
+    map2 =  sunpy.map.GenericMap(arr_2, header2)
+    header3 = make_fitswcs_header(arr_3, ref_coord,)
+    map3 =  sunpy.map.GenericMap(arr_3, header3)
+
+    return [map1, map2, map3]
```

Comment:
I think it would make the test code more readable to break this up into three fixtures, particularly since you have separate tests for each of them anyway. When looking at the tests with the combined fixures, I had a hard time telling what exactly you were testing in each one.

Alternatively, if you're just using each map in only a single test, you could avoid a fixture altogether and just create the map within the test. Fixtures are only really useful when you need the same object across multiple tests (e.g. your `test_granule_map`).

Also, this might be slightly pedantic but I would avoid prepending fixture names with "test_" as this is the convention used by pytest to denote test functions.

Reply:
sure, I can break them up. I had only done that to not have to recreate the header information three times.